pkg/apis/trust/v1alpha1/types_bundle.go

```diff
@@ -217,6 +217,29 @@ type KeySelector struct {
 	// Key is the key of the entry in the object's `data` field to be used.
 	// +kubebuilder:validation:MinLength=1
 	Key string `json:"key"`
+
+	// Metadata is an optional set of labels and annotations to be copied to the target.
+	// +optional
+	Metadata *TargetMetadata `json:"metadata"`
+}
+
+// KeySelectorWithoutMetadata is a reference to a key for some map data object without optional metadata
+type KeySelectorWithoutMetadata struct {
```

Comment:
Even if your suggestion would change less code, I think we should avoid changing unrelated code (additional formats). So I created an alternative solution here: https://github.com/cert-manager/trust-manager/compare/main...erikgb:trust-manager:pr-582-suggestion. Please let me know what you think!

Reply:
Ah nice, yeah this looks good, I'll pull that in

Reply:
Done. I've added some getters on TargetTemplate which handle both if Metadata or annotations/labels are nil, is the `types_bundle.go` the best place, or is it worth splitting them out into another file?